config/bases/kustomization.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - cluster-api/cluster.x-k8s.io_clusters.yaml
+  - cluster-api/cluster.x-k8s.io_machinesets.yaml
+  - cluster-api/cluster.x-k8s.io_machines.yaml
+  - cluster-api/cluster.x-k8s.io_machinedeployments.yaml
+  - cluster-api/cluster.x-k8s.io_machinehealthchecks.yaml
+  - cluster-api/manager-serviceaccount.yaml
+  - cluster-api/manager-clusterrole.yaml
+  - cluster-api/manager-clusterrolebinding.yaml
+  - cluster-api/manager-deployment.yaml
+
+  - hypershift-operator/hypershift.openshift.io_openshiftclusters.yaml
+  - hypershift-operator/hypershift.openshift.io_hostedcontrolplanes.yaml
+  - hypershift-operator/hypershift.openshift.io_guestclusters.yaml
+  - hypershift-operator/hypershift.openshift.io_nodepools.yaml
+  - hypershift-operator/operator-namespace.yaml
+  - hypershift-operator/operator-serviceaccount.yaml
+  - hypershift-operator/operator-clusterrole.yaml
+  - hypershift-operator/operator-clusterrolebinding.yaml
+  - hypershift-operator/operator-deployment.yaml
+
+commonLabels:
+  cluster.x-k8s.io/v1alpha4: v1alpha1
```

Comment:
just to clarify as a side effect of this new structure this label will be a applied to a bunch of manifests unnecessarily, the only ones actually requiring it are `hostedcontrolplanes` and `guestclusters`

Reply:
To clarify, they should be applied to the customresourcedefinition resources themselves?

Reply:
yep, this label is to set the contract between cluster.x-k8s.io and the external references https://cluster-api.sigs.k8s.io/developer/providers/v1alpha2-to-v1alpha3.html#apply-the-contract-version-label-clusterx-k8sioversion-version1_version2_version3-to-your-crds

Reply:
Thanks, I'm refactoring this now

Reply:
Fixed this so that the labels are applied only to the relevant CRDs using a targeted patch